src/Moat/Pretty/Doc/Markdown.hs

```diff
@@ -0,0 +1,139 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+module Moat.Pretty.Doc.Markdown
+  ( block,
+    br,
+    document,
+    inline,
+    inlineContent,
+    li,
+    markdown,
+    node,
+    ol,
+    para,
+    parseDoc,
+    text,
+    ul,
+  )
+where
+
+import CMarkGFM
+import Documentation.Haddock.Markup
+import Documentation.Haddock.Parser
+import Documentation.Haddock.Types
+import qualified Data.Text as T
+
+-- | Parse a Haddock comment.
+--
+-- See 'markdown' to generate Markdown nodes from this value.
+parseDoc :: String -> DocH m String
+parseDoc docstr = toRegular $ _doc $ parseParas Nothing docstr
+
+-- | Translate Haddock documentation into Markdown nodes.
+markdown ::
+  -- | Function to format a module name as a Markdown node.
+  (String -> Node) ->
+  -- | Function to format an identifier as a Markdown node.
+  (String -> Node) ->
+  -- | Incoming Haddock documentation.
+  DocH String String
+  -- | Generated markdown nodes.
+  -> Node
```

Comment:
ahhhh, that's kinda surprising that you can't put doc comments on the return!

Reply:
TIL...